src/main/java/domain/WinningLotto.java

```diff
@@ -0,0 +1,40 @@
+package domain;
+
+import java.util.List;
+
+public class WinningLotto {
+	private final Lotto winnigLotto;
+	private final LottoBall bonusNumber;
+
+	public WinningLotto(Lotto winnigLotto, LottoBall bonusNumber) {
+		if (winnigLotto.contains(bonusNumber)) {
+			throw new IllegalArgumentException("보너스 번호는 당첨 번호와 중복되면 안됩니다.");
+		}
+		this.winnigLotto = winnigLotto;
+		this.bonusNumber = bonusNumber;
+	}
+
+	public WinningMoney getResultAmount(Lotto lotto) {
+		return getRank(lotto).getWinningMoney();
+	}
+
+	public Rank getRank(Lotto lotto) {
+		int matchNumberCount = countSameNumber(lotto);
+		if (matchNumberCount != 5) {
+			return Rank.findByMatchNumberCount(matchNumberCount);
+		}
+		if (hasBonusNumber(lotto)) {
+			return Rank.MATCH_FIVE_WITH_BONUS;
+		}
+		return Rank.MATCH_FIVE;
+	}
+
+	private int countSameNumber(Lotto lotto) {
+		List<LottoBall> targetLottoNumbers = lotto.getLottoNumbers();
+		return (int) targetLottoNumbers.stream().filter(winnigLotto::contains).count();
```

Comment:
아래와 같이 lotto 객체에게 [메시지를 보내](https://tecoble.techcourse.co.kr/post/2020-04-28-ask-instead-of-getter/)보면 어떨까요? 
```suggestion
		return winnigLotto.countSameNumber(lotto);
```

Reply:
모든 로또에게 필요하다기보단 `WinningLotto`에서만 필요하다 생각했습니다. 근데 `Lotto`에서 해당 역할을 수행할 수 있도록 하면 아예 지금 이 메소드는 필요 없겠네요! 삭제하고 `getRank()`에 옮겼습니다.